### DISCUSSION TITLE
Partial derivative

### DISCUSSION THREAD

**Discussion by Peter230655** [sympy/sympy#27337] [Q&A]
Sa,y I have a sympy expression E, which contains functions of t, and maybe also t explicitly, e.g. E = sm.cos(x(t)) + sm.exp(x(t) * t)
To see, whether t is in the expression explicitly, I would check whether $\dfrac{\partial E}{\partial t} == 0$
I could not find anything in the sympy documentation. Did I miss it, or is it not available?
**Thanks for any help!**

**Comment by oscarbenjamin** [comment]
I think it is not available. The `diff` function already computes "partial derivatives" but it is not what you meant. You can use:
```python
In [15]: from sympy.core.function import AppliedUndef

In [16]: x = Function('x')

In [17]: E = cos(x(t)) + exp(x(t) * t)

In [18]: E2 = E.replace(lambda e: isinstance(e, AppliedUndef), lambda e: Dummy('rep_'+str(e)))

In [19]: E2
Out[19]: 
 repₓ₍ₜ₎⋅t               
ℯ          + cos(repₓ₍ₜ₎)

In [20]: E2.has(t)
True
```

**Comment by Peter230655** [comment]
**Thanks a lot for this solution!**
